sdks/go/pkg/beam/core/runtime/exec/sdf_invokers.go

```diff
@@ -351,3 +352,97 @@ func (n *cweInvoker) Reset() {
 		n.args[i] = nil
 	}
 }
+
+// trInvoker is an invoker for TruncateRestriction.
+type trInvoker struct {
+	fn   *funcx.Fn
+	args []interface{}
+	call func(rest interface{}, elms *FullValue) (pair interface{})
+}
+
+var offsetrangeTracker = reflect.TypeOf((*offsetrange.Tracker)(nil)).Elem()
+
+func DefaultTruncateRestriction(restTracker interface{}) (newRest interface{}) {
+	tracker, ok := restTracker.(sdf.BoundableRTracker)
+	if !ok {
+		return nil
+	}
+	switch tracker.(type) {
+	case *offsetrange.Tracker:
+		if tracker.(*offsetrange.Tracker).IsBounded() {
+			return restTracker.(*offsetrange.Tracker).GetRestriction().(offsetrange.Restriction)
+		}
+	default:
+		return nil
+	}
+	return nil
+}
+
+func newTruncateRestrictionInvoker(fn *funcx.Fn) (*trInvoker, error) {
+	n := &trInvoker{
+		fn:   fn,
+		args: make([]interface{}, len(fn.Param)),
+	}
+	if err := n.initCallFn(); err != nil {
+		return nil, errors.WithContext(err, "sdf TruncateRestriction invoker")
+	}
+	return n, nil
+}
+
+func newDefaultTruncateRestriction() (*trInvoker, error) {
```

Comment:
Nit: this should probably be `newDefaultTruncateRestrictionInvoker` for consistency